sycl/include/CL/sycl/detail/sycl_mem_obj_t.hpp

```diff
@@ -51,7 +52,13 @@ class __SYCL_EXPORT SYCLMemObjT : public SYCLMemObjI {
 
   template <typename T>
   using EnableIfOutputIteratorT = enable_if_t<
-      /*is_output_iterator<T>::value &&*/ !std::is_pointer<T>::value>;
+      /*is_output_iterator<T>::value &&*/ !std::is_pointer<T>::value &&
+      !std::is_same<typename T::value_type, bool>::value>;
```

Comment:
@alexbatashev, @sergey-semenov, could you, please, help reviewing this change?

Reply:
I'm afraid this is going to break some use cases: https://godbolt.org/z/nznfq1

Reply:
There is the test SYCL/buffer/buffer.cpp . There are a lot of checks, including pointer (for ex int*, void*). The only compilation error was the error with void*, because there was an iterator_value_type_t<T> and void type was trying to convert to void&. With typename T::value_type all the checks were passed

Reply:
Ok, sounds good to me